erpnext/stock/get_item_details.py

```diff
@@ -669,6 +669,9 @@ def get_price_list_rate_for(args, item_code):
 			item_price_args["uom"] = args.get("stock_uom")
 			general_price_list_rate = get_item_price(item_price_args, item_code, ignore_party=args.get("ignore_party"))
 
+		if not general_price_list_rate:
+			general_price_list_rate = get_item_price(item_price_args, item_code, ignore_party=True)
+
```

Comment:
Due to line **662**, in the following `get_item_price` , the condition at line **626** is considered since there is no **supplier** or **customer**.  And it fetches the item price anyway. 
So it fails at line **656**  and tries at **665** where it is successful. 
If customer is not set it will be NULL in the db. 
Under what condition will it reach this attempt of fetching the Item Price ?

Reply:
Before raising this PR, I was also unsure about the cause. After debugging the same issue thrice, i found that, customer was set to NULL but the same item price which is supposed to be fetch has supplier set as something. 

So the conditions which says that, 
`(customer = '' or customer is null) AND (supplier = '' or supplier is null)` fails the item price to be fetched.

Will change the `get_item_price()` query and remove the current changes.